### PR TITLE
[ADD] estate: Accept and refuse offer functions T#70778

### PR DESCRIPTION
**Pull request by Gilmaciasr**
In this one i added buttons to change property status with restriction, if a property is sold it cannot change to canceled and viceversa, buttons to accept or refuse an offer were added on offer tab, when accept an offer the selling price and buyer are set  automatically a property cannot have 2 offers accepted.
Link to the lesson:
https://www.odoo.com/documentation/15.0/developer/tutorials/getting_started/10_actions.html
Video showing changes:
https://youtu.be/wAEVm_NHtNs

### CODE REVIEW

**Comment by Gilmaciasr** [comment]
@JesusValdez96  PR #4 is ready.

File: estate/views/estate_property_views.xml [hunk 1]
```diff
@@ -27,12 +27,17 @@
       <field name="model">estate.property</field>
       <field name="arch" type="xml">
         <form string="Properties">
+          <header>
+            <button name= "sold_button_action" type="object" string="Sold"/>
+            <button name= "cancel_button_action" type="object" string="Cancel"/>
```
Comment (JesusValdez96) [inline estate/views/estate_property_views.xml:32]:
```suggestion
            <button name="sold_button_action" type="object" string="Sold"/>
            <button name="cancel_button_action" type="object" string="Cancel"/>
```

File: estate/views/estate_property_offer_views.xml [hunk 2]
```diff
@@ -28,6 +28,8 @@
         <field name="partner_id"/>
         <field name="validity" string="Validity (days)"/>
         <field name="date_deadline" string="Deadline"/>
+        <button name="accept_offer_button_action" string="Accept" type="object" icon="fa-check"/>
+        <button name="refuse_offer_button_action" string="Refuse" type="object" icon="fa-check"/>
```
Comment (JesusValdez96) [inline estate/views/estate_property_offer_views.xml:32]: where can I see this? there's no property offers tree view or property offers menu item in the web page, this two views are necessary?

**Review by JesusValdez96 — CHANGES_REQUESTED** [review]
@Gilmaciasr Good job, any questions do not hesitate to ask me.

File: estate/models/estate_property_offer.py [hunk 3]
```diff
@@ -35,3 +35,13 @@ def _compute_date_deadline(self):
...
+
+    def accept_offer_button_action(self):
+        offers = self.property_id.offer_ids.mapped("status")
+        if "accepted" not in offers:
+            for record in self:
+                record.status = "accepted"
+
+    def refuse_offer_button_action(self):
+        for record in self:
+            record.status = "refused"
```
[the remark this answers is not in the document]
Comment (Gilmaciasr) [reply estate/models/estate_property_offer.py:46]: @JesusValdez96 
In the instructions of the exercise they mention that this is a good practice.
https://www.odoo.com/documentation/15.0/developer/tutorials/getting_started/10_actions.html
image

**Pull request merged**